components/reusable-components/embla-carousel-dot-button/EmblaCarouselDotButton.tsx

```diff
@@ -0,0 +1,56 @@
+import { useCallback, useEffect, useState } from 'react';
+import { EmblaCarouselType } from 'embla-carousel';
+
+type UseDotButtonType = {
+  selectedIndex: number;
+  scrollSnaps: number[];
+  // eslint-disable-next-line no-unused-vars
+  onDotButtonClick: (index: number) => void;
+};
+
+export const useDotButton = (emblaApi: EmblaCarouselType | undefined): UseDotButtonType => {
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [scrollSnaps, setScrollSnaps] = useState<number[]>([]);
+
+  const onDotButtonClick = useCallback(
+    (index: number) => {
+      if (!emblaApi) return;
+      emblaApi.scrollTo(index);
+    },
+    [emblaApi]
+  );
+
+  const onInit = useCallback((emblaApiDots: EmblaCarouselType) => {
+    setScrollSnaps(emblaApiDots.scrollSnapList());
+  }, []);
+
+  const onSelect = useCallback((emblaApiDots: EmblaCarouselType) => {
+    setSelectedIndex(emblaApiDots.selectedScrollSnap());
+  }, []);
+
+  useEffect(() => {
+    if (!emblaApi) return;
+
+    onInit(emblaApi);
+    onSelect(emblaApi);
+    emblaApi.on('reInit', onInit);
```

Comment:
@ninman27 we can use embaApi.off() in the return function of the `useEffect` in order to do a cleanup?